New blog post on backup frequency (#3133)
alcionai/corso: website/blog/2023-04-24-backup-frequency.md

---
slug: how-often-should-you-run-microsoft-365-backups
title: "How often should you run Microsoft 365 backups?"

> description: "On the ideal cadence for backups"

Reviewer: Can you add a slightly longer description, please?

Author: extended description

authors: nica
tags: [corso, microsoft 365, backups, best practices]
date: 2023-04-24
image: ./images/astro-clock.jpg
---
<!-- vale Vale.Spelling = NO -->
![a closeup of the Prague Astronomical Clock By EWilson (Volunteer) - Own work, CC BY-SA 4.0, https://commons.wikimedia.org/w/index.php?curid=115416822](./images/astro-clock.jpg)
<!-- vale Vale.Spelling = YES -->

> I was inspired by some recent conversations with Corso users on Discord, and

Reviewer: Link to discord

Author: added link

state tokens to show what has changed about your data. For example Corso only
captures incremental changes during backups, only needing to store the items
that have been updated or added since the last backup. It does this using
[state tokens](https://learn.microsoft.com/en-us/graph/delta-query-overview#state-tokens)
that it stores within the Microsoft 365 infrastructure to checkpoint the end of
a backup. This token is used by the next backup invocation to see what has
changed, including deletions, within your data.

The exact expiry of state tokens isn’t published by Microsoft, but our
observations show that if you are only backing up every few days, these tokens
can expire. This will force a new full backup each time which is both
unnecessary and costly (in terms of time and bandwidth but not
storage because of Corso’s deduplication).

You can therefore reduce data transmission overhead, improve backup performance, and reduce RPO, by backing up more frequently.

## Cost Considerations: Storage Costs

With the threat of ransomware and other malicious data corruption, it’s a great
idea to store full backups with some frequency. This means that, if you want to

> have frequent backups ****with**** retention of older versions, you’re going to

Reviewer: ```suggestion
have frequent backups **with** retention of older versions, you’re going to
```

> need a lot of storage.  

Reviewer: ```suggestion
need a lot of storage unless your backup system is smart.  
```

> Tools like Corso that use S3 object storage will be cheaper than most others.

Reviewer: ```suggestion
Intelligent tools like Corso will be cheaper than most others. First, it uses object storage which
is orders of magnitude cheaper than reliable block or file storage.
```

Further Corso not only deduplicates, packs, and compresses data, but it also has
a ton of smarts to only capture incremental changes between backups but always
present them as a full backup (topic for another blog post!). This ensures that
even if you have 1000s of backups per user or SharePoint site, you will always
see fast restores, minimal storage overhead, and always-full backups.

For other tools, you should evaluate if it uses storage efficiently:

- Since there’s a per-object storage cost with most S3 tiers, backups should bundle small items together
- Backups should include compression and de-duplication to be as small as possible

> Take a look at some of our recent writing on selecting the best S3 storage tier

Reviewer: Link needed

Author: added link

(spoiler warning it’s probably Glacier IR) for your S3 backups.

### You still haven’t answered my question: How often should you back up?

Independent of whether it's Microsoft 365 or other systems, at least once a

> day. Probably about once every 8 hours. It will ensure your backups are
> incremental and that you don’t lose too much work in the event of an incident.

Reviewer: ```suggestion
day. Probably about once every 8 hours. It will ensure your backups are
efficient due to incremental data capture and that you don’t lose too much work in the event of an incident.
```
